arch/arm64/core/Kconfig

```diff
@@ -11,6 +11,15 @@ config CPU_CORTEX_A
 	help
 	  This option signifies the use of a CPU of the Cortex-A family.
 
+config CPU_CORTEX_R_AARCH64
```

Comment:
I wonder if having the `_AARCH64` suffix here still makes sense given that now (supposedly) AArch32 and AArch64 namespaces are separated.

Reply:
I tried to remove _AARCH64, but some aarch32 tests (cmsis..?) are using CONFIG_COTEX_R as a filter. So what is your suggestion?

Reply:
Also `subsys/testsuite/include/test_asm_inline_gcc.h` needs to do some seperate work too?

Reply:
In that case we should have `CONFIG_CORTEX_R && CONFIG_ARM` for ARM and `CONFIG_CORTEX_R && CONFIG_ARM64` for AArch64 to differentiate. @ioannisg what's do you think about this?

Reply:
What's more, for example, zephyr/tests/lib/cmsis_dsp/distance/testcase.yaml, `filter: ((CONFIG_CPU_CORTEX_R or CONFIG_CPU_CORTEX_M) and TOOLCHAIN_HAS_NEWLIB == 1) or CONFIG_ARCH_POSIX`.
There are 10+ test cases having these kind of filter

Reply:
Yeah. Alright, let this like it is now with the suffix. I'll discuss with @ioannisg if we need a nomenclature change there but it's matter for another PR.

Reply:
I think we do need something here.
How about CPU_AARCH64_CORTEX_R

Reply:
On the other hand, would that be identical to ARMV8_R? If so, we could skip it.

Reply:
Sorry for the delay. `CPU_AARCH64_CORTEX_R` is totally OK for me, I can change it.
As the `ARMV8_R`, I am not clear with what to skip by using `ARMV8_R`, do you mean those test cases or something?
Or use ARMV8_R instead of CPU_AARCH64_CORTEX_R and remove the CPU_AARCH64_CORTEX_R  totally? @ioannisg

Reply:
For now, ARMV8_R is identical but may cause the same kind of problems in the future. Because Cortex-R52 is an Armv8r aarch32 processor which is not introduced yet.